Jenkinsfile

```diff
@@ -331,22 +332,53 @@ pipeline {
               }
             }
           }
-          stage('Release to NuGet') {
-            options { skipDefaultCheckout() }
+          stage('Release') {
+            options {
+              skipDefaultCheckout()
+              timeout(time: 12, unit: 'HOURS')
+            }
             when {
-              beforeAgent true
+              beforeInput true
               anyOf {
-                tag "\\d+\\.\\d+\\.\\d+*"
+                tag pattern: 'v\\d+\\.\\d+\\d+', comparator: 'REGEXP'
                 expression { return params.Run_As_Master_Branch }
               }
             }
-            steps {
-              input(message: 'Should we release a new version on NuGet?', ok: 'Yes, we should.')
-              withGithubNotify(context: 'Release NuGet', tab: 'artifacts') {
-                deleteDir()
-                unstash 'source'
-                dir("${BASE_DIR}"){
-                  release('secret/apm-team/ci/elastic-observability-nuget')
+            stages {
+              stage('Notify') {
+                steps {
+                    emailext subject: '[apm-agent-dotnet] Release ready to be pushed',
+                            to: "${NOTIFY_TO}",
+                            body: "Please go to ${env.BUILD_URL}input to approve or reject within 12 hours."
+                }
+              }
+              stage('Release to NuGet') {
```

Comment:
Soo... to be honest we never really used this stage - we usually manually deployed the packages (I can give more info on why... the point is: this is a little bit untested :) )

I think ideally there would be 2 nuget feeds that we use:
- One for every daily build (or on every commit to `master` - I'm fine with anything) - this would be an unofficial release mainly for internal usage - this could be AppVeyor - or something else.
- And one for the official releases - this would be nuget.org and we need to be careful with this one - we only should push official releases there.

Now, to your question:
>  Clarify with the apm-agent-dotnet whether the Release to AppVeyor stage should only happen when there is commit into master and nothing when there is a tag release.

Yes, that sounds good to me - if I understand correctly with that, we'd test all commit to master, no reason to test a tag release in my opinion. - But the important point (and this is the reason for my rant above: this should go into AppVayor (or some other nuget feed) and not into nuget.org). Then I'd handle the "official releases" differently.

I also tried to understand what this stage exactly does, and I'm a little bit unsure. The `def release(secret){` part that this uses calls `.ci/linux/deploy.sh`, which does a `nuget push`. But that pushes to a nuget feed that is configured somewhere -or uses the default, which is nuget.org.

So... I think we should discuss to which nuget feed we want to push for this (probably not nuget.org) and configure accordingly.

Reply:
Just had a chat with @gregkalapos. This particular PR will delete the section regarding the NuGet release as it's not used yet. Then we can think about the release automation based on the above requirements. There will be some changes in the upcoming weeks to define this particular release process.

To keep it simple, this particular PR cares for all the tags that matches the `X.Y.Z` pattern, then, it will bump the agent version of the opbeans-dotnet and create a tag in that particular repo that matches the tag in this repo.

The current release process does create the tagged release in this repo after the artifacts have been uploaded to NuGet. There is also a requirement of matching the tagged releases with the version of the apm-agent-dotnet that has been uploaded to NuGet.